cmake/Findllvm.cmake

```
@@ -67,47 +67,53 @@ NEED_TARGET("XCore")
 
 if(NOT(CMAKE_BUILD_TYPE STREQUAL "Debug") OR ZIG_STATIC)
   execute_process(
-      COMMAND ${LLVM_CONFIG_EXE} --libfiles --link-static
-      OUTPUT_VARIABLE LLVM_LIBRARIES_SPACES
-      OUTPUT_STRIP_TRAILING_WHITESPACE)
-  string(REPLACE " " ";" LLVM_LIBRARIES "${LLVM_LIBRARIES_SPACES}")
+    COMMAND ${LLVM_CONFIG_EXE} --libdir --link-static
+    OUTPUT_VARIABLE LLVM_LIBDIR_ORIG
+    OUTPUT_STRIP_TRAILING_WHITESPACE)
+  file(TO_CMAKE_PATH ${LLVM_LIBDIR_ORIG} LLVM_LIBDIR)
 
   execute_process(
-      COMMAND ${LLVM_CONFIG_EXE} --system-libs --link-static
-      OUTPUT_VARIABLE LLVM_SYSTEM_LIBS_SPACES
-      OUTPUT_STRIP_TRAILING_WHITESPACE)
-  string(REPLACE " " ";" LLVM_SYSTEM_LIBS "${LLVM_SYSTEM_LIBS_SPACES}")
+    COMMAND ${LLVM_CONFIG_EXE} --libnames --link-static
+    OUTPUT_VARIABLE LLVM_LIBNAMES_SPACES
+    OUTPUT_STRIP_TRAILING_WHITESPACE)
+  string(REPLACE " " ";" LLVM_LIBNAMES "${LLVM_LIBNAMES_SPACES}")
+  foreach(LLVM_LIBNAME ${LLVM_LIBNAMES})
+    list(APPEND LLVM_LIBRARIES ${LLVM_LIBDIR}/${LLVM_LIBNAME})
+  endforeach()
 
   execute_process(
-      COMMAND ${LLVM_CONFIG_EXE} --libdir --link-static
-      OUTPUT_VARIABLE LLVM_LIBDIRS_SPACES
-      OUTPUT_STRIP_TRAILING_WHITESPACE)
-  string(REPLACE " " ";" LLVM_LIBDIRS "${LLVM_LIBDIRS_SPACES}")
+    COMMAND ${LLVM_CONFIG_EXE} --system-libs --link-static
+    OUTPUT_VARIABLE LLVM_SYSTEM_LIBS_SPACES
+    OUTPUT_STRIP_TRAILING_WHITESPACE)
+    string(REPLACE " " ";" LLVM_SYSTEM_LIBS "${LLVM_SYSTEM_LIBS_SPACES}")
 endif()
 if(NOT LLVM_LIBRARIES)
   execute_process(
-      COMMAND ${LLVM_CONFIG_EXE} --libs
-      OUTPUT_VARIABLE LLVM_LIBRARIES_SPACES
-      OUTPUT_STRIP_TRAILING_WHITESPACE)
-  string(REPLACE " " ";" LLVM_LIBRARIES "${LLVM_LIBRARIES_SPACES}")
+    COMMAND ${LLVM_CONFIG_EXE} --libdir
+    OUTPUT_VARIABLE LLVM_LIBDIR_ORIG
+    OUTPUT_STRIP_TRAILING_WHITESPACE)
+  file(TO_CMAKE_PATH ${LLVM_LIBDIR_ORIG} LLVM_LIBDIR)
 
   execute_process(
-      COMMAND ${LLVM_CONFIG_EXE} --system-libs
-      OUTPUT_VARIABLE LLVM_SYSTEM_LIBS_SPACES
-      OUTPUT_STRIP_TRAILING_WHITESPACE)
-  string(REPLACE " " ";" LLVM_SYSTEM_LIBS "${LLVM_SYSTEM_LIBS_SPACES}")
+    COMMAND ${LLVM_CONFIG_EXE} --libs
+    OUTPUT_VARIABLE LLVM_LIBNAMES_SPACES
+    OUTPUT_STRIP_TRAILING_WHITESPACE)
+    string(REPLACE " " ";" LLVM_LIBNAMES "${LLVM_LIBNAMES_SPACES}")
+  foreach(LLVM_LIBNAME ${LLVM_LIBNAMES})
+    list(APPEND LLVM_LIBRARIES ${LLVM_LIBDIR}/${LLVM_LIBNAME})
```

```suggestion
    list(APPEND LLVM_LIBRARIES "${LLVM_LIBDIR}/${LLVM_LIBNAME}")
```

in cmake you have to quote the shit out of everything